lib/components/narrative/default/default-itinerary.js

```diff
@@ -263,68 +287,82 @@ class DefaultItinerary extends NarrativeItinerary {
         role="presentation"
       >
-        <button
+        <ItinerarySummaryWrapper
           className="header"
           // _onHeaderClick comes from super component (NarrativeItinerary).
           onClick={this._onHeaderClick}
         >
-          {/* FIXME: semantics - replace the divs with span (we are inside a button) */}
-          <ItinerarySummaryWrapper>
-            <div className="title">
+          <div className="title">
+            <h3>
               <ItineraryDescription intl={intl} itinerary={itinerary} />
-              <ItinerarySummary itinerary={itinerary} LegIcon={LegIcon} />
-              {itineraryHasAccessibilityScores(itinerary) && (
-                <AccessibilityRating
-                  gradationMap={localizedGradationMapWithIcons}
-                  large
-                  score={getAccessibilityScoreForItinerary(itinerary)}
-                />
-              )}
-            </div>
-            {isFlexItinerary && (
-              <FlexIndicator
-                isCallAhead={isCallAhead}
-                isContinuousDropoff={isCoordinationRequired}
-                phoneNumber={phone}
+            </h3>
+            <ItinerarySummary itinerary={itinerary} LegIcon={LegIcon} />
+            {itineraryHasAccessibilityScores(itinerary) && (
+              <AccessibilityRating
+                gradationMap={localizedGradationMapWithIcons}
+                large
+                score={getAccessibilityScoreForItinerary(itinerary)}
               />
             )}
-            <ul className="list-unstyled itinerary-attributes">
-              <FieldTripGroupSize itinerary={itinerary} />
-              {ITINERARY_ATTRIBUTES.sort((a, b) => {
-                const aSelected = this._isSortingOnAttribute(a)
-                const bSelected = this._isSortingOnAttribute(b)
-                if (aSelected) return -1
-                if (bSelected) return 1
-                return a.order - b.order
-              }).map((attribute) => {
+          </div>
+          {isFlexItinerary && (
+            <FlexIndicator
+              isCallAhead={isCallAhead}
+              isContinuousDropoff={isCoordinationRequired}
+              phoneNumber={phone}
+            />
+          )}
+          <ul
+            aria-label="itinerary details"
+            className="list-unstyled itinerary-attributes"
+          >
+            <FieldTripGroupSize itinerary={itinerary} />
+            {ITINERARY_ATTRIBUTES.sort((a, b) => {
+              const aSelected = this._isSortingOnAttribute(a)
+              const bSelected = this._isSortingOnAttribute(b)
+              if (aSelected) return -1
+              if (bSelected) return 1
+              return a.order - b.order
+            })
+              .filter(
+                (x) =>
+                  x.render(
+                    itinerary,
+                    itineraryAttributeOptions,
+                    defaultFareKey
+                  ) !== undefined
```

Comment:
Can we reuse these rendering results on line 346?

Reply:
Good idea! Added in c388b25ddda47344657e8b72a354a43e1b69a9c2 thanks!!